### PR TITLE
Cherry-pick PR #416 to release-v1.2

### PR DESCRIPTION
**Pull request by thunderboltsid**
### Cherry-Pick Details
- **Original PR Title:** Explicitly set a exponential backoff rate limiter for controller config
- **Original PR URL:** https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/416
- **Commit SHA:** 714c151c10fcb1b6c26031a028198eaf063349c9
- **Cherry-Pick Reason:** Support configuring the rate limiter and configure narrower defaults to reduce number of requests to IAM stack

### CODE REVIEW

**Review by adiantum — APPROVED** [review]
/lgtm
/approve

File: main.go [hunk 1]
```diff
@@ -99,6 +97,11 @@ func main() {
 		TimeEncoder: zapcore.RFC3339TimeEncoder,
 	}
 	opts.BindFlags(flag.CommandLine)
+
+	flag.DurationVar(&baseDelay, "rate-limiter-base-delay", 500*time.Millisecond, "The base delay for the rate limiter.")
+	flag.DurationVar(&maxDelay, "rate-limiter-max-delay", 15*time.Minute, "The maximum delay for the rate limiter.")
```
Comment (deepakm-ntnx) [inline main.go:102]: nit: ideally would prefer a unit in the name itself.

**Review by deepakm-ntnx — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (thunderboltsid) [reply main.go:102]: why is `rate-limiter-base-delay-seconds=0.5` or `rate-limiter-base-delay-seconds=1` prefereable over
`rate-limiter-base-delay=500ms` or `rate-limiter-base-delay=1s` ? The latter is a more standard and versatile way of admitting input.

**Pull request merged**